### PR TITLE
25966 metrics v51bundle

### LINKED ISSUE #25966: Create/Update MP Metrics 5.1 bundles/features w.r.t to MP Metrics 5.1.0-RC1
1. Create the `io.openliberty.microprofile.metrics.5.1.internal` bundle **with updated** `MetricRegistryProducerAdapter`
MP Metrics 5.1 has updated the `@RegistryScope` annotation to be a qualifier.

This will require an updated `MetricRegistryProducerAdapter` to facilitate this change.
This will be placed in the  `io.openliberty.microprofile.metrics.5.1.internal` bundle which is yet to be created. This issue will also cover the creation of this bundle.

Due to minimal changed between v5.1 and v5.0, this bundle will re-export all the packages + classes in the `io.openliberty.microprofile.metrics.5.0.internal` bundle.
However, to ensure the `MetricRegistryProducerAdapter` from the`io.openliberty.microprofile.metrics.5.1.internal` is loaded properly, changed will be made to the `LegacyMetricsExtensionAdapter` class. This class explicitly defines classes to be treated as CDI beans by the CDI runtime. We will attempt to the load the v5.1 `MetricRegistryProducerAdapter` using reflection. Failing that, it will load the v5.0  `MetricRegistryProducerAdapter`.

See: https://github.com/eclipse/microprofile-metrics/pull/778

2. Create `metrics.5.1.bnd`  (i.e. `io.openliberty.org.eclipse.microprofile.metrics.5.1` bundle -name) under `io.openliberty.org.eclipse.microprofile` and pull in 5.1.0-RC1

3. Update `restfulWS3.1-mpMetrics-monitor1.0.feature` to include `mpMetrics-5.1`'

4. Update `restfulWSDefaultExceptionMapper-1.0.feature` to include `mpMetrics-5.1`

5. Update maven to `5.1.0-RC1` in `io.openliberty.org.eclipse.microprofile.metrics-5.1.feature` and point to the `io.openliberty.org.eclipse.microprofile.metrics.5.1` bundle

6. Update the `mpMetrics-5.1` feature to point to the `io.openliberty.microprofile.metrics.5.1.internal` bundle

### PR DESCRIPTION
**Pull request by Channyboy**
#build
Fixes : #25966

### CODE REVIEW

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/LegacyMetricsExtensionAdapter.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-2.0/
- *
```
Comment (pgunapal) [inline dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/LegacyMetricsExtensionAdapter.java:7]: Update Copyright year.

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-2.0/
- *
```
Comment (pgunapal) [inline dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java:7]: Update Copyright year

File: dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/MetricRegistryProducerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
```
Comment (pgunapal) [inline dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/MetricRegistryProducerAdapter.java]: Update Copyright year

File: dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/package-info.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
```
Comment (pgunapal) [inline dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/package-info.java]: Update Copyright year.

File: dev/io.openliberty.org.eclipse.microprofile/metrics.5.1.bnd [hunk 5]
```diff
@@ -0,0 +1,35 @@
+#*******************************************************************************
```
Comment (pgunapal) [inline dev/io.openliberty.org.eclipse.microprofile/metrics.5.1.bnd:1]: Fix copyright year, should be 2023

**Review by pgunapal — CHANGES_REQUESTED** [review]
New files need to have 2023 as the copyright year.
Updated files should have 2023 added as the copyright year.

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java (hunk 2 above)
Comment (Channyboy) [reply dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java:7]: file has spacing changes, copyright not needed

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/LegacyMetricsExtensionAdapter.java (hunk 1 above)
Comment (Channyboy) [reply dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/cdi/adapters/LegacyMetricsExtensionAdapter.java:7]: fixed

File: dev/io.openliberty.org.eclipse.microprofile/metrics.5.1.bnd (hunk 5 above)
Comment (Channyboy) [reply dev/io.openliberty.org.eclipse.microprofile/metrics.5.1.bnd:1]: fixed

File: dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/MetricRegistryProducerAdapter.java (hunk 3 above)
Comment (Channyboy) [reply dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/MetricRegistryProducerAdapter.java]: fixed

File: dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/package-info.java (hunk 4 above)
Comment (Channyboy) [reply dev/io.openliberty.microprofile.metrics.5.1.internal/src/io/openliberty/smallrye/metrics/cdi/adapters51/package-info.java]: fixed

**Review by pgunapal — APPROVED** [review]
LGTM

**Comment by Channyboy** [comment]
#libby

**Comment by Channyboy** [comment]
#libby-bot

**Comment by Channyboy** [comment]
#libby

**Pull request merged**